src/TBuffer.cpp

```diff
@@ -3092,6 +3110,7 @@ void TBuffer::append(const QString& text, int sub_start, int sub_end, TChar form
     if (length < 1) {
         return;
     }
+    length = std::min(length, MAX_CHARACTERS_PER_ECHO);
```

Comment:
Here as well: should we let the user know that we were unable to fulfil the request fully? Or should we fail with an error? Or should we split the echo if it's larger than allowed?

Reply:
I'm not sure if that'll be helpful to them. We already log weird symbols and emojis and in a way it's just spamming stderr...

Reply:
Won't they wonder why only half of their content was echoed?

Reply:
If they echoed 20,000 characters in a single line? :|

Reply:
Who knows? But if you say it's okay, leave it be :smile:

Reply:
I think it's alright, because showing an error won't help them too much anyway. It would also seem like modern mobile/web apps are doing away with throwing errors at the user and instead attempt to recover the best they can, or simply not work, and you get something went wrong from the behaviour of the app anyhow.